internal/checker/emitresolver.go

```diff
@@ -901,11 +901,10 @@ func (r *emitResolver) CreateLiteralConstValue(emitContext *printer.EmitContext,
 		return nil // non-literal type
 	}
 	literalValue := t.AsLiteralType().value
-	switch literalValue.(type) {
+	switch value := literalValue.(type) {
 	case string:
 		return emitContext.Factory.NewStringLiteral(literalValue.(string))
 	case jsnum.Number:
```

Comment:
This one's still redundant; you can just reuse the name "literalValue" and the linter should point these out.